src/Modules/OrchardCore.Commerce/Services/SessionShoppingCartPersistence.cs

```diff
@@ -45,4 +45,12 @@ protected override async Task<bool> StoreInnerAsync(string key, ShoppingCart ite
 
         return true;
     }
+
+    protected override async Task<bool> RemoveInnerAsync(string key)
+    {
+        await Task.Run(Session.Clear, _httpContextAccessor.HttpContext.RequestAborted);
```

Comment:
Why is the `Task.Run` necessary? Seems like unnecessary overhead to me. You can remove the `async` from this method and just call `Session.Clear()` here.